subsys/net/ip/tcp2.c

```diff
@@ -804,6 +811,11 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
 		goto next_state;
 	}
 
+	if (!tcp_options_check((th + 1), tcp_options_len)) {
+		NET_ERR("DROP: Invalid TCP option list");
```

Comment:
@rmstoi 

This NET_ERR would be extra, since there's one added in tcp_options_check()